test/tests/functional/pbs_checkpoint.py

```diff
@@ -0,0 +1,179 @@
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+from ptl.utils.pbs_crayutils import CrayUtils
+
+
+class TestCheckpoint(TestFunctional):
+    """
+    This test suite targets Checkpoint functionality.
+    """
+    abort_file = ''
+    cu = CrayUtils()
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        abort_script = """#!/bin/bash
+kill $1
+exit 0
+"""
+        self.abort_file = self.du.create_temp_file(body=abort_script)
+        self.du.chmod(path=self.abort_file, mode=0755)
+        c = {'$action': 'checkpoint_abort 30 !' + self.abort_file + ' %sid'}
+        self.mom.add_config(c)
+        self.platform = self.du.get_platform()
+        if self.platform != 'cray' and self.platform != 'craysim':
+            self.attrs = {ATTR_l + '.select': '1:ncpus=1',
+                          ATTR_l + '.place': 'excl'}
+        else:
+            nv = self.cu.num_compute_vnodes(self.server)
+            self.attrs = {ATTR_l + '.select': '%d:ncpus=1' % nv,
+                          ATTR_l + '.place': 'scatter'}
+
+    def verify_checkpoint_abort(self, jid, stime):
+        """
+        Verify that checkpoint and abort happened.
+        """
+        self.ck_dir = os.path.join(self.server.pbs_conf['PBS_HOME'],
+                                   'checkpoint', jid + '.CK')
+        self.assertTrue(os.path.isdir(self.ck_dir),
+                        msg="Checkpoint directory " + self.ck_dir +
+                        " not found")
+        _msg1 = "%s;req_holdjob: Checkpoint initiated." % jid
+        self.mom.log_match(_msg1, max_attempts=3, starttime=stime)
+        _msg2 = "%s;checkpoint_abort script %s: exit code 0" % (
+            jid, self.abort_file)
+        self.mom.log_match(_msg2, max_attempts=3, starttime=stime)
+        _msg3 = "%s;checkpointed to %s" % (jid, self.ck_dir)
+        self.mom.log_match(_msg3, max_attempts=3, starttime=stime)
+        _msg4 = "%s;task 00000001 terminated" % jid
+        self.mom.log_match(_msg4, max_attempts=3, starttime=stime)
```

Comment:
For all of the above log_match, Why max_attempts=3?  I've seen tests that try to limit the max_attempts to a small number, fail on a machine that is running slow.

Reply:
I agree, we should remove max_attempts and use the default.

Reply:
Removed max_attempts.